dnf5/commands/list/list.cpp

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "list.hpp"
+
+#include "utils/bgettext/bgettext-mark-domain.h"
+
+#include <libdnf-cli/output/package_list_sections.hpp>
+#include <libdnf/rpm/package_query.hpp>
+
+#include <iostream>
+
+namespace dnf5 {
+
+using namespace libdnf::cli;
+
+
+void ListCommand::set_parent_command() {
+    auto * arg_parser_parent_cmd = get_session().get_argument_parser().get_root_command();
+    auto * arg_parser_this_cmd = get_argument_parser_command();
+    arg_parser_parent_cmd->register_command(arg_parser_this_cmd);
+    arg_parser_parent_cmd->get_group("query_commands").register_argument(arg_parser_this_cmd);
+}
+
+void ListCommand::set_argument_parser() {
+    auto & ctx = get_context();
+    auto & parser = ctx.get_argument_parser();
+    auto & config = ctx.base.get_config();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_description("Lists packages depending on the packages' relation to the system");
+
+    auto specs = parser.add_new_positional_arg("specs", ArgumentParser::PositionalArg::UNLIMITED, nullptr, nullptr);
+    specs->set_description("List of keys to match");
+    specs->set_parse_hook_func(
+        [this]([[maybe_unused]] ArgumentParser::PositionalArg * arg, int argc, const char * const argv[]) {
+            for (int i = 0; i < argc; ++i) {
+                pkg_specs.emplace_back(argv[i]);
+            }
+            return true;
+        });
+    specs->set_complete_hook_func([&ctx](const char * arg) { return match_specs(ctx, arg, true, true, false, false); });
+    cmd.register_positional_arg(specs);
+
+    auto showduplicates = parser.add_new_named_arg("showduplicates");
+    showduplicates->set_long_name("showduplicates");
+    showduplicates->set_description("Show all versions of the packages, not only the latest ones.");
+    showduplicates->set_const_value("true");
+    showduplicates->link_value(&config.showdupesfromrepos());
```

Comment:
Well, we can drop the config option and keep only commandline option for list / info / search commands.
It shouldn't be a problem given it's not documented in dnf4.
But I'd prefer to do it in separate PR - it will affect also the `search` command.